aroquev/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,14 +36,24 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxComments = Integer.parseInt(request.getParameter("maxComments"));
+
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    final ArrayList<String> comments = new ArrayList<String>();
-    for (Entity entity : results.asIterable()) {
-      String comment = (String) entity.getProperty("commentBody");
+    final ArrayList<Comment> comments = new ArrayList<Comment>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxComments))) {
+      long id = entity.getKey().getId();
```

Comment:
It might be nice to create a Comment factory that can take an Entity comment and create a Comment? Could be used here and a few other places.

Reply:
+1 🏭 
And/or use the Builder pattern so you don't have a constructor with a bunch of parameters.